libs/ledgerjs/packages/context-module/src/loaders/NftLoader.ts

```diff
@@ -0,0 +1,72 @@
+import axios from "axios";
+import { LoaderOptions } from "../models/LoaderOptions";
+import { Transaction } from "../models/Transaction";
+import { ContextLoader } from "./ContextLoader";
+import { ContextResponse } from "../models/ContextResponse";
+
+enum ERC721_SUPPORTED_SELECTOR {
+  Approve = "0x095ea7b3",
+  SetApprovalForAll = "0xa22cb465",
+  TransferFrom = "0x23b872dd",
+  SafeTransferFrom = "0x42842e0e",
+  SafeTransferFromWithData = "0xb88d4fde",
+}
+
+enum ERC1155_SUPPORTED_SELECTOR {
+  SetApprovalForAll = "0xa22cb465",
+  SafeTransferFrom = "0xf242432a",
+  SafeBatchTransferFrom = "0x2eb2c2d6",
+}
+
+const SUPPORTED_SELECTORS: `0x${string}`[] = [
+  ...Object.values(ERC721_SUPPORTED_SELECTOR),
+  ...Object.values(ERC1155_SUPPORTED_SELECTOR),
+];
+
+export class NftLoader implements ContextLoader {
+  constructor() {}
+
+  async load(transaction: Transaction, _options: LoaderOptions) {
+    const responses: ContextResponse[] = [];
+
+    if (!transaction.to || !transaction.data || transaction.data === "0x") {
+      return [];
+    }
+
+    const selector = transaction.data.slice(0, 10) as `0x${string}`;
+
+    if (!this.isSelectorSupported(selector)) {
+      return [];
+    }
+
+    // EXAMPLE:
+    // https://nft.api.live.ledger.com/v1/ethereum/1/contracts/0xBC4CA0EdA7647A8aB7C2061c2E118A18a936f13D/plugin-selector/0x095ea7b3
+    const pluginResponse = await axios.request<{ payload: string }>({
```

Comment:
[SHOULD] Isolate Axios call into a class like `NftHttpApiClient`. 
It will facilitate the unit testing as for me `NftLoader`is like a repository/service so at domain level and Http Call should be done at a data level.

Reply:
Since all calls from different loaders are done with Axios, It is not better to create just an `HttpApiClient` and use it in all current loaders ?

Reply:
Just sharing a personal feeling regarding this: It's a very Ledger thingy to always abstract everything everywhere, and it can make the code very hard to read (at least for people used to javascript). I feel like we should limit abstractions to things where it adds a specific value to the project.

Reply:
no abstraction needed here but Single Responsability principle, we can gather all HTTPCall in one client in that way it would be a lot easier to test the loader.

Reply:
Done 👍